### PR TITLE
Change thumbnail src to thumbnail id for lightbox thumbnail api

### PR DESCRIPTION
**Pull request by cathyxz**
Closes https://github.com/ampproject/amphtml/issues/12856. 
Pending this PR, `<amp-lightbox-gallery>` API should be feature complete. =) 

TODOs: a default icon from Material Design was used here for testing purposes, that will be substituted with the launch icons per https://github.com/ampproject/amphtml/issues/12713.

### CODE REVIEW

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-placeholders.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const defaultPlaceholder = 'data:image/svg+xml;utf8,<svg fill="#FFFFFF" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="M6 2c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6H6zm7 7V3.5L18.5 9H13z"/><path d="M0 0h24v24H0z" fill="none"/></svg>';
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/service/lightbox-placeholders.js]: This file will eventually be replaced by actual icons.

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js [hunk 2]
```diff
@@ -368,27 +372,66 @@ export class LightboxManager {
...
+      const thumbnailImage = element.ownerDocument.getElementById(thumbnailId);
+      user().assert(thumbnailImage.tagName == 'AMP-IMG');
+      return srcsetFromElement(thumbnailImage);
     } else if (element.tagName == 'AMP-IMG') {
-      return element.getAttribute('src');
+      return srcsetFromElement(element);
+    } else if (element.tagName == 'AMP-ANIM') {
+      return srcsetFromElement(element);
+    } else if (element.tagName == 'AMP-VIDEO') {
+      return this.getThumbnailSrcsetForVideo_(element);
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js]: We'll add more if this doesn't address all supported elements.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 3]
```diff
@@ -1080,7 +1081,12 @@ export class AmpLightboxGallery extends AMP.BaseElement {
     element.classList.add('i-amphtml-lbg-gallery-thumbnail');
     const imgElement = this.win.document.createElement('img');
     imgElement.classList.add('i-amphtml-lbg-gallery-thumbnail-img');
-    imgElement.setAttribute('src', thumbnailObj.url);
+
+    if (thumbnailObj.srcset) {
+      imgElement.setAttribute('srcset', thumbnailObj.srcset.stringify());
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: TODO: pending `sizes` and `srcset` fix, we can revisit the logic here to make sure this works properly with them.

Comment (cvializ) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: I think if both `src` and `srcset` are present on an image, both are semantically meaningful. I think here you want to apply each to `imgElement`, not just either one. Here's an example of both being used in the spec: http://w3c.github.io/html/semantics-embedded-content.html#example-15076cde

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js [hunk 4]
```diff
@@ -368,27 +372,66 @@ export class LightboxManager {
...
    * @param {!Element} element
-   * @param {number=} index fake it for testing only, will delete later
    * @return {string}
    * @private
    */
-  getThumbnailUrl_(element, index) {
-    if (element.hasAttribute('lightbox-thumbnail-src')) {
-      return element.getAttribute('lightbox-thumbnail-src');
+  getPlaceholderForElementType_(element) {
+    switch (element.tagName) {
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js]: nit: Closure might automagically convert this already, but this should be simplified to a single return statement until the switch syntax is actually used when #12713 is implemented.

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js [hunk 5]
```diff
@@ -368,27 +372,66 @@ export class LightboxManager {
...
     }
   }
+
+  /**
+   * Given an amp video, returns the thumbnail srcset.
+   * @param {!Element} ampVideo
+   * @return {!../../../../src/srcset.Srcset|null}
+   */
+  getThumbnailSrcsetForVideo_(ampVideo) {
+    if (ampVideo.hasAttribute('poster')) {
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js]: optional style nit: to simplify the code usually we use ternary syntax for simple `if-else`s like this.

```js
const poster = ampVideo.getAttribute('poster');
return poster ? srcsetFromSrc(poster) : null;
```

It also eliminates the call to `hasAttribute`.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js (hunk 3 above)
Comment (cathyxz) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: The intention here is that if the user provided a `srcset` or we can generate a `srcset` from any kind of placeholder, then we use `srcset`. However, if the user did not provide a `srcset`, then we use our default placeholders. I originally wanted to only use `srcset`, however data uri with svgs don't work with `srcset`. Hence if a `srcset` is ever provided, we don't want to be using `src`. Maybe it is more appropriate if I put a comment here.

Comment (cvializ) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: Oh I see, `srcsetFromElement` converts the `src` to `srcset` if needed. 👍

**Pull request merged**